internal/cel/kuadrant_validator.go

```diff
@@ -0,0 +1,109 @@
+package cel
+
+import (
+	"github.com/google/cel-go/cel"
+
+	"github.com/kuadrant/kuadrant-operator/internal/wasm"
+)
+
+const (
+	AuthPolicyKind           = "AuthPolicy"
+	RateLimitPolicyKind      = "RateLimitPolicy"
+	TokenRateLimitPolicyKind = "TokenRateLimitPolicy"
+
+	AuthPolicyName = "auth"
+	RateLimitName  = "ratelimit"
+)
+
+var StateCELValidationErrors = "CELValidationErrors"
+
+type Issue struct {
+	policyKind string
+	pathID     string
+	err        error
+}
+
+func NewIssue(action wasm.Action, pathID string, err error) *Issue {
+	return &Issue{
+		policyKind: policyKindFromWasmServiceName(action.ServiceName),
+		pathID:     pathID,
+		err:        err,
+	}
+}
+
+func (i *Issue) GetError() error {
+	return i.err
+}
+
+type IssueCollection struct {
+	issues map[string]map[string][]*Issue
```

Comment:
I'm not strongly opinionated here but I wonder if the clarity in just having this as a `[]*Issue` where your `GetByPolicyKind` does a `lo.Filter` on `issue.policyKind` instead of this several layer nested structure could be more readable, wdyt?